pkg/pillar/base/logobjecttypes.go

```diff
@@ -0,0 +1,443 @@
+func EnsureLogObject(objType LogObjectType, objName string, objUUID uuid.UUID, key string) *LogObject {
+	logObject := LookupLogObject(key)
+	if logObject == nil {
+		logObject = NewLogObject(objType, objName, objUUID, key)
+	}
+	return logObject
+}
+
+// DeleteLogObject :
+func DeleteLogObject(key string) {
+	_, ok := logObjectMap[key]
+	if !ok {
+		log.Errorf("DeleteLogObject: LogObject with key %s not found in internal map", key)
+		return
+	}
+	delete(logObjectMap, key)
+}
+
+// AddField : Add a key value pair to be logged
+func (object *LogObject) AddField(key string, value interface{}) *LogObject {
+	object.Fields[key] = value
+	return object
+}
+
+// AddCompositeField : Add a structure/map to be logged
+func (object *LogObject) AddCompositeField(key string, value interface{}) (*LogObject, error) {
+	b, err := json.Marshal(value)
+	if err != nil {
+		return nil, err
+	}
+	itemMap := make(map[string]interface{})
+	err = json.Unmarshal(b, &itemMap)
+	if err != nil {
+		return nil, err
+	}
+	object.AddFields(itemMap)
+	return object, nil
+}
+
+// AddFields : Values of exiting keys in the object will be overwritten with new values passed
+func (object *LogObject) AddFields(fields map[string]interface{}) *LogObject {
+	for key, value := range fields {
+		object.Fields[key] = value
+	}
+	return object
+}
+
+// Merge :
+// Values of existing fields in destination object will be overwritten with values
+// from source object.
+func (object *LogObject) Merge(source *LogObject) *LogObject {
+	for key, value := range source.Fields {
+		object.Fields[key] = value
+	}
+	return object
+}
+
+// Clone : Create a clone from an existing Log object
+func (object *LogObject) Clone() *LogObject {
+	newLogObject := new(LogObject)
+	newLogObject.Fields = make(map[string]interface{})
+	for key, value := range object.Fields {
+		newLogObject.Fields[key] = value
+	}
+	newLogObject.Initialized = true
+	return newLogObject
+}
+
+// CloneAndAddField : Add key value pair to a cloned log object
+func (object *LogObject) CloneAndAddField(key string, value interface{}) *LogObject {
+	newLogObject := object.Clone()
+	newLogObject.AddField(key, value)
+	return newLogObject
+}
+
+// CloneAndAddCompositeField : Add composite structure to a cloned log object
+func (object *LogObject) CloneAndAddCompositeField(key string, value interface{}) *LogObject {
+	newLogObject := object.Clone()
+	newLogObject.AddCompositeField(key, value)
+	return newLogObject
+}
+
+// CloneAndAddFields : Add additional fields to a cloned log object
+func (object *LogObject) CloneAndAddFields(fields map[string]interface{}) *LogObject {
+	newLogObject := object.Clone()
+	newLogObject.AddFields(fields)
+	return newLogObject
+}
+
+// CloneAndMerge : Merge source into a cloned log object
+func (object *LogObject) CloneAndMerge(source *LogObject) *LogObject {
+	newLogObject := object.Clone()
+	newLogObject.Merge(source)
+	return newLogObject
+}
+
+// Debug :
```

Comment:
How about putting these boilerplate level functions (Debug etc) in a separate file?

Reply:
will do